agent/consul/leader_connect_ca.go

```diff
@@ -1548,3 +1546,16 @@ func (c *CAManager) checkExpired(pem string) error {
 	}
 	return nil
 }
+
+func primaryUsesIntermediate(provider ca.Provider) bool {
+	_, ok := provider.(ca.PrimaryUsesIntermediate)
+	return ok
+}
+
+func (c *CAManager) isIntermediateUsedToSignLeaf() bool {
+	if c.serverConf.Datacenter != c.serverConf.PrimaryDatacenter {
+		return true
+	}
```

Comment:
so if we are not the primary DC we always sign with the intermediate?

Reply:
Yup, exactly! Every provider uses an intermediate certificate in a secondary DC to sign leaf certs.